File: benchmark/recurse.lua
-- Recursive calls and repeated stack unwinding.
local function recurse(n)
    if n == 0 then
        return 0
    end
    return recurse(n - 1) + 1
end

local t0 = os.clock()
local i = 0
while i < 1000 do
    recurse(1000)
    i = i + 1
end

print((os.clock() - t0) * 1000)

File: benchmark/recurse.py
# Recursive calls and repeated stack unwinding.
import sys
import time


sys.setrecursionlimit(3000)


def recurse(n):
    if n == 0:
        return 0
    return recurse(n - 1) + 1


t0 = time.perf_counter()
i = 0
while i < 1000:
    recurse(1000)
    i += 1

print((time.perf_counter() - t0) * 1000)
